contracts/rate_oracles/GlpRateOracle.sol

```diff
@@ -130,27 +130,26 @@ contract GlpRateOracle is BaseRateOracle, IGlpRateOracle {
     {
         // get last observation
         uint216 lastIndexRay = observations[oracleVars.rateIndex].observedValue;
-        require(lastIndexRay > 0, "No previous observation");
 
         // calculate rate increase since last update
         uint256 cumulativeRewardPerToken = rewardTracker
             .cumulativeRewardPerToken();
+
+        if (cumulativeRewardPerToken < lastCumulativeRewardPerToken) {
+            revert CustomErrors.GlpRewardTrackerUnorderedRate();
+        }
+
         uint256 rewardsRateSinceLastUpdate = FullMath.mulDiv(
             cumulativeRewardPerToken - lastCumulativeRewardPerToken,
             lastEthPriceInGlp,
             GLP_PRECISION
         ); // GLP_PRECISION
 
         // compute index using rate increase & last index
-        resultRay = FullMath.mulDiv(
-            GLP_PRECISION + rewardsRateSinceLastUpdate,
-            lastIndexRay,
-            GLP_PRECISION
-        );
+        resultRay = lastIndexRay + rewardsRateSinceLastUpdate / 1000;
```

Comment:
can you remind me why we need to divide by 1000? can we use a library to do the conversion or alternatively just represent 1000 as some constant to improve readability?

Reply:
It's because `rewardsRateSinceLastUpdate` has GLP_PRECISION (1e30). I introduced a nicer conversion [c2a5c2f](https://github.com/Voltz-Protocol/voltz-core/pull/230/commits/c2a5c2fda8ffd7f8ede64ea6b3b1d51a822d32d0)